lib/presentation/auth/change_password/confirm/bloc/change_password_confirm_cubit.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_template/data/auth/model/auth_data_error.dart';
+import 'package:flutter_template/data/auth/repository/auth_repository.dart';
+import 'package:flutter_template/data/auth/service/remote/model/forgot_password_confirm/forgot_password_confirm_request_model.dart';
+import 'package:flutter_template/presentation/auth/change_password/confirm/bloc/change_password_confirm_error.dart';
+import 'package:flutter_template/presentation/auth/change_password/confirm/bloc/change_password_confirm_state.dart';
+import 'package:flutter_template/util/extensions/string_extension.dart';
+import 'package:flutter_template/util/mixins/resettable_bloc_mixin.dart';
+
+class ChangePasswordConfirmCubit extends Cubit<ChangePasswordConfirmState>
+    with ResettableBlocMixin {
+  final AuthRepository _authRepository;
+
+  ChangePasswordConfirmCubit(this._authRepository)
+      : super(
+          const ChangePasswordConfirmStateInitial(
+            email: "",
+            password: "",
+            confirmPassword: "",
+            code: "",
+          ),
+        );
+
+  @override
+  void resetState() {
+    emit(
+      ChangePasswordConfirmStateInitial(
+        email: state.email,
+        password: state.password,
+        confirmPassword: state.confirmPassword,
+        code: state.code,
+      ),
+    );
+  }
```

Comment:
Somethinig handy to avoid this, just add an `.initial()` or `.empty()` `factory constructor` on the model to reuse the this code.